feat: deprecate mongo db like QL syntax in redis and support native QL (#645)
jina-ai/docarray: docs/advanced/document-store/redis.md

| `ef_construction` | Optional parameter for Redis HNSW algorithm                                                       | `200`                                             |
| `m`               | Optional parameter for Redis HNSW algorithm                                                       | `16`                                              |
| `ef_runtime`      | Optional parameter for Redis HNSW algorithm                                                       | `10`                                              |
| `block_size`      | Optional parameter for Redis FLAT algorithm                                                       | `1048576`                                         |
| `initial_cap`     | Optional parameter for Redis HNSW and FLAT algorithm                                              | `None`, defaults to the default value in Redis    |
| `columns`         | Other fields to store in Document and build schema                                                | `None`                                            |

> You can check the default values in [the docarray source code](https://github.com/jina-ai/docarray/blob/main/docarray/array/storage/redis/backend.py)

Reviewer: ```suggestion
You can check the default values in [the docarray source code](https://github.com/jina-ai/docarray/blob/main/docarray/array/storage/redis/backend.py).
```

> For vector search configurations, default values are those of the database backend, you can find them in [redis documentation](https://redis.io/docs/stack/search/reference/vectors/)

Reviewer: ```suggestion
For vector search configurations, default values are those of the database backend, which you can find in the [Redis documentation](https://redis.io/docs/stack/search/reference/vectors/).
```

```console
Embeddings Approximate Nearest Neighbours with "price" at most 7, "color" blue and "stock" True:

 score=12,	 embedding=[6. 6. 6.],	 price=6,	 color=blue,	 stock=1
 score=48,	 embedding=[4. 4. 4.],	 price=4,	 color=blue,	 stock=1
 score=108,	 embedding=[2. 2. 2.],	 price=2,	 color=blue,	 stock=1
 score=192,	 embedding=[0. 0. 0.],	 price=0,	 color=blue,	 stock=1
```

````{admonition} Note
:class: note
Note that Redis does not support Boolean types in attributes. Therefore, you need to configure your boolean field as 
integer in `columns` configuration (`'field': 'int'`) and use a filter query that treats the field as an integer
(`@field: [1 1]`).
````

### Search by filter query

> One can search with user-defined query filters using the `.find` method. Such queries follow the [Redis' Search Query Syntax](https://redis.io/docs/stack/search/reference/query_syntax/).

Reviewer: ```suggestion
One can search with user-defined query filters using the `.find` method. Such queries follow the [Redis Search Query Syntax](https://redis.io/docs/stack/search/reference/query_syntax/).
```

> Consider you store Documents with a certain tag `price` into Redis and you want to retrieve all Documents

Reviewer: ```suggestion
Consider a case where you store Documents with a tag of `price` into Redis and you want to retrieve all Documents
```

> with `price`  lower or equal to  some `max_price` value.

Reviewer: ```suggestion
with `price` less than or equal to  some `max_price` value.
```

You can index such Documents as follows:

```python
from docarray import Document, DocumentArray

n_dim = 3
da = DocumentArray(
    storage='redis',
    config={
        'n_dim': n_dim,
        'columns': {'price': 'float'},
    },
)

with da:
    da.extend([Document(id=f'r{i}', tags={'price': i}) for i in range(10)])

print('\nIndexed Prices:\n')
for price in da[:, 'tags__price']:
    print(f'\t price={price}')
```

> Then you can retrieve all documents whose price is lower than or equal to `max_price` by applying the following

Reviewer: ```suggestion
Then you can retrieve all documents whose price is less than or equal to `max_price` by applying the following
```
